pkg/cmd/eventlistener/describe.go

```diff
@@ -0,0 +1,198 @@
+
+    tkn eventlistener describe foo -n bar
+
+or
+
+   tkn el desc foo -n bar
+`
+
+	c := &cobra.Command{
+		Use:     "describe",
+		Aliases: []string{"desc"},
+		Short:   "Describes a eventlistener in a namespace",
+		Example: eg,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Args:         cobra.MinimumNArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+
+			if err := validate.NamespaceExists(p); err != nil {
+				return err
+			}
+
+			output, err := cmd.LocalFlags().GetString("output")
+			if err != nil {
+				fmt.Fprint(os.Stderr, "Error: output option not set properly \n")
+				return err
+			}
+
+			if output != "" {
+				return describeEventListenerOutput(cmd.OutOrStdout(), p, f, args[0])
+			}
+
+			return printEventListenerDescription(s, p, args[0])
+		},
+	}
+
+	_ = c.MarkZshCompPositionalArgumentCustom(1, "__tkn_get_eventlistener")
+	f.AddFlags(c)
+	return c
+}
+
+func describeEventListenerOutput(w io.Writer, p cli.Params, f *cliopts.PrintFlags, name string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	el, err := cs.Triggers.TriggersV1alpha1().EventListeners(p.Namespace()).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// NOTE: this is required for -o json|yaml to work properly since
+	// tektoncd go client fails to set these; probably a bug
+	el.GetObjectKind().SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: "triggers.tekton.dev",
+			Kind:    "EventListener",
+		})
+
+	return printer.PrintObject(w, el, f)
+}
+
+func printEventListenerDescription(s *cli.Stream, p cli.Params, ttname string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return fmt.Errorf("failed to create tekton client")
+	}
+
+	el, err := cs.Triggers.TriggersV1alpha1().EventListeners(p.Namespace()).Get(ttname, metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintf(s.Err, "failed to get eventlistener %s\n", ttname)
+		return err
+	}
+
+	var data = struct {
+		EventListener *v1alpha1.EventListener
+	}{
+		EventListener: el,
+	}
+
+	funcMap := template.FuncMap{
+		"decorate":                formatted.DecorateAttr,
+		"getEventListenerTrigger": getEventListenerTrigger,
+		"getURL":                  getURL,
+		"getEventListenerName":    getEventListenerName,
+	}
+
+	w := tabwriter.NewWriter(s.Out, 0, 5, 3, ' ', tabwriter.TabIndent)
+	tparsed := template.Must(template.New("Describe EventListener").Funcs(funcMap).Parse(describeTemplate))
+	if err = tparsed.Execute(w, data); err != nil {
+		fmt.Fprintf(s.Err, "Failed to execute template \n")
+		return err
+	}
+	return w.Flush()
+}
+
+func getEventListenerTrigger(listener v1alpha1.EventListener) string {
+	var out bytes.Buffer
+	resourceTemplate, err := json.Marshal(listener.Spec.Triggers)
+	if err != nil {
+		log.Printf("json marshal failed with error: %v", err)
+		return ""
```

Comment:
Returning an error here would be preferred/would make this easier to test.

Reply:
Handled Please take a look